src/app/form-generator/components/node-field/node-field.component.ts

```diff
@@ -12,40 +15,118 @@ import { BaseFieldComponent } from '../base-field/base-field.component';
+
+        if (this.userValue) {
+            const node$ = this.entities.selectNode(this.userValue!, { language: this.node.language });
+
+            node$.subscribe(node => {
+                this.schemaName = node.schema.name;
+                this.isContainer = node.container;
+                this.displayName = node.displayName!;
+                this.breadcrumbPath = node.project.name + node.breadcrumb.map(b => b.displayName).join(' › ');
+            });
+
+            this.editorEffects.loadNode(this.node.project.name!, this.userValue, this.node.language);
+        }
     }
 
     valueChange(value: NodeField): void {
         this.userValue = (value && value.uuid) || '';
     }
 
     isValidUuid(): boolean {
-        return this.userValue.length === 32;
+        if (this.userValue) {
+            return this.userValue.length === 32;
+        } else {
+            return false;
+        }
     }
 
     selectNode(): void {
-        const node = this.api.getNodeValue() as MeshNode;
+        this.node = this.api.getNodeValue() as MeshNode;
+
+        const allowedSchemas = this.api.field.allow;
+
         this.api
             .openNodeBrowser({
-                startNodeUuid: node.parentNode ? node.parentNode.uuid : node.uuid,
-                projectName: node.project.name!,
-                titleKey: 'editor.select_node'
+                startNodeUuid: this.node.parentNode ? this.node.parentNode.uuid : this.node.uuid,
+                projectName: this.node.project.name!,
+                titleKey: 'editor.select_node',
+                selectablePredicate: node => {
+                    if (allowedSchemas) {
+                        return allowedSchemas.indexOf(node.schema.name) > -1;
+                    }
+                }
             })
-            .then(uuids => {
-                this.userValue = uuids[0];
-                this.api.setValue({ uuid: this.userValue });
+            .then((results: PageResult[]) => {
+                this.userValue = results[0].uuid;
+
+                if (this.userValue) {
+                    const node$ = this.entities.selectNode(this.userValue!, { language: this.node.language });
+
+                    node$.subscribe(node => {
+                        this.schemaName = node.schema.name;
+                        this.isContainer = node.container;
+                        this.displayName = node.displayName!;
+                        this.breadcrumbPath = node.project.name + node.breadcrumb.map(b => b.displayName).join(' › ');
+                        this.api.setValue({
+                            uuid: node.uuid
+                        });
+                        this.changeDetector.detectChanges();
+                    });
+
+                    this.editorEffects.loadNode(this.node.project.name!, this.userValue, this.node.language);
+                }
                 this.changeDetector.detectChanges();
             });
     }
+
+    editNode(): void {
```

Comment:
We just need this line in valueChange():
`this.routerLink = this.navigationService.detail(this.node.project.name!, this.userValue!, this.node.language).commands();`.
And I also removed a click handler from the template because we don't need it.